### PR TITLE
Saga's Saga

### PR DESCRIPTION
**Pull request by SpearDane**
New Mission String in Human Space

## Summary

The primary goal of this PR is to add greater depth to human space, and a companion for the player.

This PR accomplishes these goals by adding multiple new missions, and small mission strings to multiple human worlds. 

These missions are all connected to your new companion, the dog Saga, who acts as the instigator for them. 

The missions are not only spread out in space, but [also in time](https://github.com/endless-sky/endless-sky/pull/6234#issuecomment-914494484). Years pass and someone you helped once might need help again, or might be in a place to help you. Or you may run into them or hear about them on the news. The idea is that time does pass, relationships change, and the world moves on. This also means that a player might have new reasons to return to human space, or find new surprises in human space long after they've done all the main missions there. 

I also try to flesh out some of the worlds, building off their descriptions to make visits to them slightly more impactful. 

This mission set starts on New Kansas where you find Saga. 

For testing reasons you'll want to alter the wait times between events - as due to the very nature of this PR some take quite a while to fire.

Saga has her own one off events, but she also finds missions for the player as they travel around space with her.  If a player does not like the one off events, I've included a 'training' quest that should turn them off, so she only activates for the main mission strings. 

For players who've completed the Skadenga quest line there are also some added interactions. 

## Artwork Checklist
(If any artwork was added or changed by this PR, the following must be provided.)
 - [ ] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: {{insert PR link}}
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}

### CODE REVIEW

**Comment by bene-dictator** [comment]
*eyes pop\*

**Comment by alexrovw** [comment]
Can't wait to review this!

File: data/human/outfits.txt [hunk 1]
```diff
@@ -430,6 +430,37 @@ outfit "Tear of Skade"
 	"unique" 1
 	description "A strange glowing stone gifted to you by the Skadenga. It lacks any power source, yet is quite cold."
 
+	outfit "Tear of Skade (Harmonized)"
```
Comment (The-Legendary-M) [inline data/human/outfits.txt:433]: Remove tab here

**Review by The-Legendary-M — CHANGES_REQUESTED** [review]
Just a syntax thing

File: data/human/sagassaga.txt [hunk 2]
```diff
@@ -0,0 +1,3060 @@
...
+					goto owner
+				`	(Shoo the dog away.)`
+			`	Saga clearly wants to follow you, her savior, wherever you're headed. But after yelling at her a few times she finally drops are tail and slinks away into the shrubbery. Last thing you need is some lost mutt following you around. You ignore the sound of her low whimpers as you return to the spaceport.`
+				decline
+			label owner
+			`	Saga happily follows you inside the spaceport. No one seems to mind a mutt at your heels, and after a little searching you find an information kiosk. A man with tired eyes, and a wool vest stands within it.`
+			choice
+				`	(Inquire about the dog, and her possible owner.)`
+			`	The man's eyes seem to sag as you speak. Your earnest words lulling him to sleep. Once you're done he frowns, and shakes his head slowly.`
+			`	"Ser," he says with forced politeness, "this is a spaceport, not a shelter. We are aware of that stray dog. She has been prowling around for weeks. No one has come to collect her, and no one wants her. Frankly, we assumed the feral sheep would get to her by now. It would have done us all a favor.`
```
Comment (bene-dictator) [inline data/human/sagassaga.txt:43]: This line will need to be put into `.codespell.exclude`.

**Review by bene-dictator — COMMENTED** [review]
Some preliminary stuff, I'll give a better look in the future.

File: data/human/sagassaga.txt [hunk 3]
```diff
@@ -0,0 +1,3060 @@
...
+				`	(Try to find Saga's owner.)`
+					goto owner
+				`	(Shoo the dog away.)`
+			`	Saga clearly wants to follow you, her savior, wherever you're headed. But after yelling at her a few times she finally drops are tail and slinks away into the shrubbery. Last thing you need is some lost mutt following you around. You ignore the sound of her low whimpers as you return to the spaceport.`
+				decline
+			label owner
+			`	Saga happily follows you inside the spaceport. No one seems to mind a mutt at your heels, and after a little searching you find an information kiosk. A man with tired eyes, and a wool vest stands within it.`
+			choice
+				`	(Inquire about the dog, and her possible owner.)`
+			`	The man's eyes seem to sag as you speak. Your earnest words lulling him to sleep. Once you're done he frowns, and shakes his head slowly.`
```
Comment (lumbar527) [inline data/human/sagassaga.txt:42]: It's not clear to me what's happening here. Do you mean this:
```suggestion
			`	The man's eyes seem to sag as you speak; your earnest words seem to be lulling him to sleep. Once you're done speaking he frowns, and shakes his head slowly.`
```

File: data/human/sagassaga.txt (hunk 2 above)
Comment (lumbar527) [inline data/human/sagassaga.txt:43]: To make the spellcheck stop complaining I think you need to add the whole line (with indents) to the `codespell.exclue` file.

**Review by lumbar527 — COMMENTED** [review]
First mission looks good; will get to the others soon.

File: data/human/outfits.txt (hunk 1 above)
Comment (bene-dictator) [reply data/human/outfits.txt:433]:
```suggestion
outfit "Tear of Skade (Harmonized)"
```

File: data/human/sagassaga.txt [hunk 4]
```diff
@@ -0,0 +1,3073 @@
...
+				decline
+			label agree
+			action
+				event "welandwait2"
+			`	Weland beams. "That's the spirit, now. Figurin' things out, well, that's what it means to be human, isn't it? Now, as I said, I reckon it'll take me near about 6 months to get this beauty figured out. I realize, being a fancy space hero an' all, you probably will need a little reminder, so when I am done I'll send you a message. As I promised, if I can't get it to work, you don't have to pay me a penny. But, well, wouldn't be betting on that happening, I haven't yet met a machine I couldn't crack."`
+			decline
+
+event "welandwait2" 183
+
+mission "Large Bone 5"
```
Comment (lumbar527) [inline data/human/sagassaga.txt:1960]: Just a bookmark for where I stopped.

**Comment by Arachi-Lover** [comment]
I don't think missions `"dog quarg"` and `"dog heliarch"` are appropriate for the game, nor do they seem to add much to this mission set.

In the case of the Quarg, it just comes out of nowhere that they apparently inherently avoid dogs, and weirder still is it that they just take the dog because of reasons. It also is odd to have them call the dog a distraction when it's said they don't even seem aware that they're moving to avoid Saga.

In the case of the Heliarchs, it just doesn't match the familiarity the Heliarchs have of human space. They do have some knowledge of human space and events there, and they certainly are familiar with the concept of pets, so they wouldn't consider dogs to be part of a human government on the same level as the Coalition species are to one another within the Heliarchy. In the sense of "forming another Coalition" they are very surprised to learn of the Hai, and make the statement then. They wouldn't freely use that term for simply having a popular pet species. They also wouldn't call others "Heliarchs", that's not a synonym or a word in their language(s) for a government officer or ruling body, it is their title, the name of their government. Furthermore, the translators wouldn't be set to translate from dog languages, their purpose is for communication between the sentient/sapient species ie the Coalition species and humanity.

**Comment by SpearDane** [comment]
> I don't think missions `"dog quarg"` and `"dog heliarch"` are appropriate for the game, nor do they seem to add much to this mission set.
> 
> In the case of the Quarg, it just comes out of nowhere that they apparently inherently avoid dogs, and weirder still is it that they just take the dog because of reasons. It also is odd to have them call the dog a distraction when it's said they don't even seem aware that they're moving to avoid Saga.
> 
> In the case of the Heliarchs, it just doesn't match the familiarity the Heliarchs have of human space. They do have some knowledge of human space and events there, and they certainly are familiar with the concept of pets, so they wouldn't consider dogs to be part of a human government on the same level as the Coalition species are to one another within the Heliarchy. In the sense of "forming another Coalition" they are very surprised to learn of the Hai, and make the statement then. They wouldn't freely use that term for simply having a popular pet species. They also wouldn't call others "Heliarchs", that's not a synonym or a word in their language(s) for a government officer or ruling body, it is their title, the name of their government. Furthermore, the translators wouldn't be set to translate from dog languages, their purpose is for communication between the sentient/sapient species ie the Coalition species and humanity.

@Arachi-Lover 
To your first point, your dog is your companion. Think of it has a companion in any another game. They often have unique dialog or unique interactions in different places. Interactions, even ones that do not really go anywhere adds something to the experience. I would love it someone more confident in writing about the Quarg or the Coalition added at least one, if not a few potential 'flavor' interactions. 

Throughout the campaign there is something, especially in the lab mission, that indicates your dog is a little special. This may manifest itself as the Quarg avoiding your dog. Why, specifically do they do it? I do not know. I agree though, that it could be cleaned up. I was thinking maybe a year after it happens your dog somehow finds her way to you. That punishes the player less for what happens in it. 

As for your remarks on the Coalition. You have communication device that talks to humans, I really don't see there being an issue with it also working on dogs - unless you specifically want there to be one. In which case that's like your opinion isn't it? The fact they can talk to humans at all given just how different any languages would be between aliens is borderline miraculous, being able to communicate with a dog doesn't seem like a stretch - again unless you make it one. I think the interaction is mildly humorous, adds a little levity, and tells you something about the dog and how she sees herself. She's someone who makes things whole, this is something you see her do throughout all her missions. 

All of that said, if you think you can fix it up and make it 'work' in your view of how things work in Coalition, I'd me more than happy to replace what I have with that. I do not like writing for the alien races, but I felt like her having zero interaction with them would be disappointing for the player. So, please, give me an interaction you're happy with it and I'll plug it in with no qualms.

**Comment by Arachi-Lover** [comment]
I understand the idea is Saga serves as a companion and these are (except for the Quarg one taking away Saga) inconsequential in terms of what happens, but at least for the Quarg and Heliarch ones I don't think they really fit the game.

For the Quarg one, why is it the Quarg would notice or care about a special pet of humanity? They laugh at the player's suggestion of nuclear weapons being a weapon of terrible destruction that'd warrant intervention, and flat out state they're not gonna take away the player's JD during FWs, but they're this bothered by a dog who's got some lab things? It's very off.

For the Heliarch one, the Coalition translators aren't magical devices that translate all. They're very sophisticated tech pieces that mesh hardware and software just like any other equipment piece. The translator being very good does not equate to it being capable of doing any translation. They're very specifically tailored to translate between languages the Coalition expects to get interaction from regularly (ie between each of the different species that reside there) or who value the knowledge they can get from interacting with their known neighbors (ie humans). Why would they bother to go through all the trouble of understanding, developing, testing and implementing the functionality to have it translate dog? Translating between different languages of different sapient species comes with the advantage of knowing there are definitive anchored concepts you can drawn mutual understanding from, such as the ideas of spacetravel, spaceships, planets, physics that'd be must-haves for reaching the point both species reached. The translators require one actually understanding the other language first, for the equivalencies in grammar and words used to be noted and coded in for the device. With dogs, they do not have that ability.
I know the idea for the mission was to be humorous, but it simply doesn't match what we know of the Heliarchs and the joke portion comes at the expense of consistency within the Coalition.

**Comment by Hecter94** [comment]
It might be a good idea to handle this project in a more modular fashion.

Each faction comes with a lot of different lore and necessary background information to understand when it comes to writing content for them; as such, it's difficult for one person to grasp every faction well.
So perhaps this PR should simply introduce Saga as a concept, add in their encounters throughout human space, keeping the scope limited to human space for now, and then follow-up PRs can work with people familiar with other factions to add more Saga interactions to those specific factions.

This way, they can be reviewed/discussed separately; concerns with one module won't block the entire thing.

I'm not a dev, so please don't just take this as something you have to do; I'm simply bringing it up as an option to consider.

**Comment by SpearDane** [comment]
@Arachi-Lover @Hecter94 

Sounds good. I'll pull the encounters with the Coalition  and the Quarg.

**Comment by bene-dictator** [comment]
I've talked with both the devs and SpearDane about how we can make this PR suitable for vanilla. It has been brought up that we already have one dog "outfit" in the game already (Puny) and adding another would be too much, and that the time spent adapting missions around Saga might be too much of a burden, especially for new alien species. Although the concept of a series of missions initiated by outfits isn't one that has been explored in ES in the past, it might be something we could try again in the future.

At the very least, we should still be able to adapt some of the missions standalone. These will be made into their own, separate PRs. They may not all be released at the same time, depending on writing and (potential) artwork.

**Pull request closed without merging**